framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1155,6 +1195,11 @@ protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
         return getBundledTypeQualifiers();
     }
 
+    /** Apply all aliases provided via -Aaliases to the type qualifiers supported */
+    public final void applyAliases() {
```

Comment:
This method currently isn't being called, right?
This would avoid that each type system calls `addAliasedTypeAnnotation(anno)` for each qualifier.
Wouldn't it be enough to make it `protected` and call this method from somewhere?

Reply:
I added it because I thought it would be nice to call it somewhere and have it be applied to all type systems. I don't think it's needed anymore with the design you suggest above.